gocourse04/main.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+const (
+	Ungulates        Types   = "Ungulates"
+	Birds            Types   = "Birds"
+	Primates         Types   = "Primates"
+	GroomingBrush    Purpose = "Grooming Brush"
+	WaterBucket      Purpose = "Water Bucket"
+	VeterinarySyring Purpose = "Veterinary Syring"
+)
+
+type Zoo struct {
+	Areas Areas
+}
+
+func (z Zoo) FindAnimalByName(name string) {
+finish:
+	for _, area := range z.Areas {
+		for sectorName, sector := range area.Sectors {
+			for _, animal := range sector.Animals {
+				if animal.Name == name {
+					fmt.Printf("Animal %s with ID %d is located in area %s, sector %s", animal.Name, animal.ID, area.Name, sectorName)
+					break finish
+				}
+			}
+		}
+	}
+}
+
+func NewZoo() *Zoo {
+	return &Zoo{}
+}
+
+type Types string
+type Area struct {
```

Comment:
Між типами також зазвичай ставиться пустий рядок:
```suggestion
type Types string

type Area struct {
```